docs/spec_draft.md

```diff
@@ -1443,6 +1445,169 @@ specification. Numeric precision is implementation-defined.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.scatter
+
+### Semantics
+
+Generate `results` which is the values of the `inputs` operand, with several
+slices at indices specified by `scatter_indices`, updated with the values in
+`updates` using `update_computation`.
+
+The following diagram shows how elements in `updates[k]` map on elements in
+`inputs[k]` using a concrete example. The diagram picks a few example
+`updated[k[` indices and explains in detail which `inputs[k]` indices they
+correspond to.
+
+<img align="center" src="spec_images/scatter.png" />
+
+More formally, for any `k`, `updates[k][update_index] =
+inputs[k][input_index]` where:
+  * `update_scatter_dims` = [`d` for `d` in `axes(updates[k])` and `d` not in
+    `update_window_dims`].
+  * `update_scatter_index` = [`update_index[d]` for `d` in
+    `update_scatter_dims`].
+  * `start_index` =
+      * `scatter_indices[bi0, ..., :, ..., biN]` where `bi` are individual
+        elements in `update_scatter_index` and `:` is inserted at the
+        `index_vector_dim` index, if `index_vector_dim` <
+        `rank(scatter_indices)`.
+      * `[scatter_indices[update_scatter_index]]` otherwise.
+  * For `do` in `axes(inputs[k])`,
+      * `full_start_index[do]` = `start_index[ds]` if
+        `do = scatter_dims_to_operand_dims[ds]`.
+      * `full_start_index[do]` = `0` otherwise.
+  * `update_window_index` = [`update_index[d]` for `d` in `update_window_dims`].
+  * `full_offset_index` = `[oi0, ..., 0, ..., oiN]` where `oi` are individual
+    elements in `update_window_index`, and `0` is inserted at indices from
+    `inserted_window_dims`.
+  * `input_index` = `add(full_start_index, full_offset_index)`.
+    If `input_index` is out of bounds for `inputs[k]`, then the behavior is
+    implementation-defined.
```

Comment:
I think this doesn't apply to `scatter`. The situation when `result_index` is out of bounds is dealt with in the paragraph below, so we should drop this sentence here.